src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SuspiciousCastFromCharToInt.cs

```diff
@@ -0,0 +1,156 @@
+            {
+                var invocation = (IInvocationOperation)context.Operation;
+
+                if (stringSplitMethods.Contains(invocation.TargetMethod))
+                {
+                    foreach (var argument in invocation.Arguments)
+                    {
+                        if (IsImplicitCharToIntConversion(argument))
+                        {
+                            context.ReportDiagnostic(argument.CreateDiagnostic(Rule));
+                        }
+                    }
+                }
+            }, OperationKind.Invocation);
+
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextStringBuilder, out var stringBuilderType))
+                return;
+
+            //  Report implicit char-to-int conversions in calls to all StringBuilder constructors
+            context.RegisterOperationAction(context =>
+            {
+                var creation = (IObjectCreationOperation)context.Operation;
+
+                if (!creation.Constructor.ContainingType.Equals(stringBuilderType, SymbolEqualityComparer.Default))
+                    return;
+
+                foreach (var argument in creation.Arguments)
+                {
+                    if (IsImplicitCharToIntConversion(argument))
+                    {
+                        context.ReportDiagnostic(argument.CreateDiagnostic(Rule));
+                    }
+                }
+            }, OperationKind.ObjectCreation);
+
+            return;
+
+            //  Local functions
+
+            bool IsImplicitCharToIntConversion(IArgumentOperation argument)
+            {
+                return argument.Value is IConversionOperation conversion &&
+                    conversion.IsImplicit &&
+                    conversion.Operand.Type.Equals(symbols.CharType, SymbolEqualityComparer.Default) &&
+                    argument.Parameter.Type.Equals(symbols.Int32Type, SymbolEqualityComparer.Default);
+            }
+        }
+
+        //  Property bag
+#pragma warning disable CA1815 // Override equals and operator equals on value types
+        internal readonly struct RequiredSymbols
+#pragma warning restore CA1815 // Override equals and operator equals on value types
+        {
+            private RequiredSymbols(ITypeSymbol charType, ITypeSymbol int32Type, ITypeSymbol stringType)
+            {
+                CharType = charType;
+                Int32Type = int32Type;
+                StringType = stringType;
+            }
+
+            public static bool TryGetSymbols(Compilation compilation, out RequiredSymbols result)
+            {
+                var charType = compilation.GetSpecialType(SpecialType.System_Char);
+                var int32Type = compilation.GetSpecialType(SpecialType.System_Int32);
+                var stringType = compilation.GetSpecialType(SpecialType.System_String);
+
+                if (charType is not null && int32Type is not null && stringType is not null)
```

Comment:
Only `GetTypeByMetadataName` can return null. `GetSpecialType` doesn't.

Reply:
@Youssef1313 What about when building mscorelib? iirc I believe I was told in the past that `GetSpecialType` can very rarely return null.